nano/node/active_transactions.cpp

```diff
@@ -34,40 +34,63 @@ void nano::active_transactions::confirm_frontiers (nano::transaction const & tra
+	auto roots_size = size ();
+	auto max_elections = (max_broadcast_queue / 4);
+	auto check_time_exceeded = std::chrono::steady_clock::now () >= next_frontier_check;
+	auto low_active_elections = roots_size < max_elections;
+	if (check_time_exceeded || (!is_test_network && low_active_elections))
 	{
-		size_t max_elections (max_broadcast_queue / 4);
+		// When the number of active elections is low increase max number of elections for setting confirmation height.
+		if (max_broadcast_queue > roots_size + max_elections)
+		{
+			max_elections = max_broadcast_queue - roots_size;
+		}
+
+		// Spend time prioritizing accounts to reduce voting traffic
+		prioritize_frontiers_for_confirmation (is_test_network ? std::chrono::milliseconds (50) : std::chrono::seconds (2));
+
 		size_t elections_count (0);
-		for (auto i (node.store.latest_begin (transaction_a, next_frontier_account)), n (node.store.latest_end ()); i != n && !stopped && elections_count < max_elections; ++i)
+		std::unique_lock<std::mutex> lock (mutex);
+		while (!priority_cementable_frontiers.empty () && !stopped && elections_count < max_elections)
 		{
-			nano::account_info info (i->second);
-			if (info.block_count != info.confirmation_height)
+			auto cementable_account = *priority_cementable_frontiers.begin ();
+			priority_cementable_frontiers.erase (priority_cementable_frontiers.begin ());
+			lock.unlock ();
+			nano::account_info info;
+			auto error = node.store.account_get (transaction_a, cementable_account.account, info);
+			release_assert (!error);
+
+			if (info.block_count > info.confirmation_height && !this->node.pending_confirmation_height.is_processing_block (info.head))
 			{
-				auto block (node.store.block_get (transaction_a, info.head));
-				if (!start (block))
+				auto block (this->node.store.block_get (transaction_a, info.head));
+				if (!this->start (block))
 				{
 					++elections_count;
 					// Calculate votes for local representatives
 					if (representative)
 					{
-						node.block_processor.generator.add (block->hash ());
+						this->node.block_processor.generator.add (block->hash ());
 					}
 				}
-				// Update next account
-				next_frontier_account = i->first.number () + 1;
 			}
+			lock.lock ();
 		}
+		lock.unlock ();
+
 		// 4 times slower check if all frontiers were confirmed
-		int fully_confirmed_factor = (elections_count <= max_elections) ? 4 : 1;
+		int fully_confirmed_factor = (elections_count < max_elections) ? 4 : 1;
 		// Calculate next check time
-		next_frontier_check = steady_clock::now () + seconds ((representative_factor * fully_confirmed_factor) / test_network_factor);
+		next_frontier_check = steady_clock::now () + (representative_factor * fully_confirmed_factor / test_network_factor);
 		// Set next account to 0 if all frontiers were confirmed
-		next_frontier_account = (elections_count <= max_elections) ? 0 : next_frontier_account;
+		if (elections_count <= max_elections)
```

Comment:
Seems not required now after https://github.com/nanocurrency/nano-node/pull/1982/files#diff-725dceebd1b7fd4fdf9ab179415dc61bR385

Reply:
Yep good spot